src/compiler/checker.ts

```diff
@@ -42554,6 +42555,35 @@ namespace ts {
 
                     if (name.kind === SyntaxKind.PropertyAccessExpression) {
                         checkPropertyAccessExpression(name, CheckMode.Normal);
+                        if (!links.resolvedSymbol) {
+                            const expressionType = checkExpressionCached(name.expression);
+                            const infos = getApplicableIndexInfos(expressionType, getLiteralTypeFromPropertyName(name.name));
+                            if (infos.length && (expressionType as any).members && (expressionType as any).members.get(InternalSymbolName.Index)) {

```

Comment:
Use something other than `any` here 🙈

Reply:
I thought I needed that, but I tested it incorrectly.